libdnf/transaction/Transformer.cpp

```diff
@@ -49,10 +49,30 @@ static const char *sql_create_tables =
 #include "sql/create_tables.sql"
     ;
 
+static const char *sql_migrate_tables_1_2 =
+#include "sql/migrate_tables_1_2.sql"
+    ;
+
+const std::string Transformer::version = "1.2";
+
 void
 Transformer::createDatabase(SQLite3Ptr conn)
 {
     conn->exec(sql_create_tables);
+    Transformer::migrateSchema(conn);
+}
+
+void
+Transformer::migrateSchema(SQLite3Ptr conn)
+{
+    // read schema version
+    SQLite3::Query query(*conn, "select value from config where key = 'version';");
+    query.step();
```

Comment:
Check the return value of the query.step() method. Call query.get only if the return value == StepResult::ROW. In other cases there is no row prepared (found) in the result.